README.md

```diff
@@ -155,6 +163,84 @@ if __name__ == '__main__':
     IOLoop.current().run_sync(main)
 ```
 
+## `notebook-http` Mode
+
+The `KernelGatewayApp.api` command line argument can be set to `notebook-http`
+to allow a properly annotated notebook to be served via HTTP endpoints. A
+notebook cell must be annotated with a single line comment at the beginning
+describing the HTTP endpoint. An example python cell is:
+
+```python
+# GET /hello/world
+print("hello world")
+```
+
+This enables an HTTP GET on `/hello/world`. The [api_intro.ipynb](https://github.com/jupyter-incubator/kernel_gateway/blob/master/etc/api_examples/api_intro.ipynb)
+provides a full example notebook for reference.
+
+The comment syntax should be supported in every language, but it is a work in
+progress to identify all supported syntaxes.
+
+### Processing requests
+
+When your cell is invoked there will be a string variable called `REQUEST`
```

Comment:
Need to point out that in languages that require variable declaration, it is the notebook author's responsibility to declare the variable. The kernel gateway will only try to assign a value to it.